Tests/SwiftGRPCNIOTests/NIOServerTestCase.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Dispatch
+import Foundation
+@testable import SwiftGRPC
+import XCTest
+
+extension Echo_EchoRequest {
+  init(text: String) {
+    self.text = text
+  }
+}
+
+extension Echo_EchoResponse {
+  init(text: String) {
+    self.text = text
+  }
+}
+
+class NIOServerTestCase: XCTestCase {
+  func makeProvider() -> Echo_EchoProvider { return EchoProvider() }
```

Comment:
IMO not worth having a function for this

Reply:
This test case is only a skeleton that gets subclassed by concrete tests; the method will be overridden by other test cases to provide other provider implementations. See the non-NIO tests.